Allow setting ACL through py-bsd

anodos325 opened the pull request:
Original version was ultimately relying on output of acl_from_text() to set the ACL.
We want to be able to set an ACL without this information. Also removed the 'name' field.

anodos325 committed:
Allow setting ACL through py-bsd
Original version was ultimately relying on output of acl_from_text() to set the ACL.
We want to be able to set an ACL without this information. Also removed the 'name' field.

skarekrow commented on bsd/acl.pyx:
```diff
@@ -259,33 +259,30 @@ cdef class ACLEntry(object):
 
     def __setstate__(self, obj):
-        if 'text' in obj:
+        if 'text' in obj and obj['text'] is not None:
```
```suggestion
        if 'text' in obj and obj.get('text'):
```
Is cleaner to me

anodos325 replied:
I'll do one better and remove it entirely. I don't see a reason to have the text output. :)

skarekrow commented on bsd/acl.pyx:
```diff
@@ -259,33 +259,30 @@ cdef class ACLEntry(object):
 
-        if 'name' in obj:
-            self.name = obj['name']
+        if 'id' in obj and obj['id'] is not None:
```
Same here

anodos325 replied:
We need an explicit check for none type here. You can have id 0. bsd.acl sets "id = None" if ae_tag is ACL_USER_OBJ, ACL_GROUP_OBJ, or ACL_EVERYONE.

sonicaj replied:
`obj.get('id') is not None`, we can drop the first if as well ; )

anodos325 committed:
Remove more unecessary items from bsd.acl

sonicaj reviewed (commented):
@anodos325 has this been leftover intentionally - https://github.com/freenas/py-bsd/blob/e543f5cadbd185d65dbea4a4d574f53ff7b9351f/bsd/acl.pyx#L132 ? I am not sure, so confirming

anodos325 committed:
Misc changes to pybsd
- Use 'with nogil' for syscalls
- Make the validation capable of handling NFS4 ACLs
- Add a couple of comments

anodos325 committed:
whitespace fixes

william-gr commented on bsd/acl.pyx:
```diff
@@ -259,53 +317,55 @@ cdef class ACLEntry(object):
 
         if 'perms' in obj:
-            for k, v in obj['perms']:
-                self.perms[k] = v
```
I think the reason it was using a for loop was to copy the dict.
Just setting the attribute you're changing the behavior if you change the dict that was passed to bsd.acl as argument.

william-gr replied:
tl;dr: dict is not a singleton
